src/frontend/js/types/Joanie.ts

```diff
@@ -204,9 +205,10 @@ export interface AbstractCourse {
   organizations: OrganizationLight[];
   title: string;
   course_runs: CourseRun[];
+  is_self_paced: boolean;
 }
 
-export type CourseLight = Pick<AbstractCourse, 'id' | 'code' | 'title'> & {
+export type CourseLight = Pick<AbstractCourse, 'id' | 'code' | 'title' | 'is_self_paced'> & {
   cover: Nullable<JoanieFile>;
 };
```

Comment:
Until this PR, the `CourseLight` type can be shared. But since you add the field `is_self_paced` you have to create a dedicated type.

Indeed, We are using `CourseLight` type to type Joanie API Responses but from Joanie the response will never return `is_self_paced` field.

IMO you should create a Course type within `types/index.ts`

Reply:
As you said, i created a Course type (and named it as `PacedCourse`) within `types/indes.ts`. It has a `courseLight` and the boolean variable `is_self_paced`.